modules/ingestor/src/service/format.rs

```diff
@@ -92,19 +97,33 @@ impl<'g> Format {
                 let curation: Curation = serde_yml::from_slice(&buffer)?;
                 loader.load(labels, curation, digests).await
             }
+            Format::CweCatalog => {
+                let loader = CweCatalogLoader::new(graph);
+                loader.load_bytes(labels, &buffer, digests).await
+            }
+        }
+    }
+
+    pub fn is_format(&self, bytes: &[u8]) -> Result<bool, Error> {
+        match self {
+            Format::OSV => Self::is_osv(bytes),
+            Format::CSAF => Self::is_csaf(bytes),
+            Format::CVE => Self::is_cve(bytes),
+            Format::SPDX => Self::is_spdx(bytes),
+            Format::CycloneDX => Self::is_cyclonedx(bytes),
+            Format::ClearlyDefined => Self::is_clearly_defined(bytes),
+            Format::CweCatalog => Self::is_cwe_catalog(bytes),
```

Comment:
Does that mean we are parsing the full catalog, just to "check" if its the type we are expecting it to be?

Maybe off topic, but I really think that this logic should be removed and the type should be provided by the user (uploader). And then just fail when it doesn't match the expected type. Instead of testing all combinations for all uploaded documents.

Reply:
> Does that mean we are parsing the full catalog, just to "check" if its the type we are expecting it to be?

No, but I share your concern...

> Maybe off topic,

Definitely 

> but I really think that this logic should be removed and the type should be provided by the user

Why?

Reply:
fwiw, I could use something more streamy to detect the XML type, like your jsonpath stuff for detectoring correct JSON formats.